client/app/services/widget.js

```diff
@@ -146,14 +146,16 @@ function WidgetFactory($http, $location, Query) {
         }
         this.queryResult = this.getQuery().getQueryResult(maxAge);
 
-        this.queryResult.toPromise()
+        return this.queryResult.toPromise()
           .then((result) => {
             this.loading = false;
             this.data = result;
+            return result;
           })
           .catch((error) => {
             this.loading = false;
             this.data = error;
+            return error;
           });
```

Comment:
I think this has been playing with me for a while (in here and in the Widget Migration).

![image](https://user-images.githubusercontent.com/3356951/66090853-86ed6400-e55b-11e9-93a0-c69c918bdaf8.png)


It was returning the queryResult before updating `this.data`, which was a weird result when I expected the widgets to render an error and they were not doing it